tsl/src/continuous_aggs/common.c

```diff
@@ -0,0 +1,1379 @@
+							   "to true.");
+		return false;
+	}
+#endif
+#endif
+	if (!query->jointree->fromlist)
+	{
+		appendStringInfoString(hint, "FROM clause missing in the query");
+		return false;
+	}
+	if (query->commandType != CMD_SELECT)
+	{
+		appendStringInfoString(hint, "Use a SELECT query in the continuous aggregate view.");
+		return false;
+	}
+
+	if (query->hasWindowFuncs)
+	{
+		appendStringInfoString(detail,
+							   "Window functions are not supported by continuous aggregates.");
+		return false;
+	}
+
+	if (query->hasDistinctOn || query->distinctClause)
+	{
+		appendStringInfoString(detail,
+							   "DISTINCT / DISTINCT ON queries are not supported by continuous "
+							   "aggregates.");
+		return false;
+	}
+
+	if (query->limitOffset || query->limitCount)
+	{
+		appendStringInfoString(detail,
+							   "LIMIT and LIMIT OFFSET are not supported in queries defining "
+							   "continuous aggregates.");
+		appendStringInfoString(hint,
+							   "Use LIMIT and LIMIT OFFSET in SELECTS from the continuous "
+							   "aggregate view instead.");
+		return false;
+	}
+
+	if (query->sortClause && !finalized)
+	{
+		appendStringInfoString(detail,
+							   "ORDER BY is not supported in queries defining continuous "
+							   "aggregates.");
+		appendStringInfoString(hint,
+							   "Use ORDER BY clauses in SELECTS from the continuous aggregate view "
+							   "instead.");
+		return false;
+	}
+
+	if (query->hasRecursive || query->hasSubLinks || query->hasTargetSRFs || query->cteList)
+	{
+		appendStringInfoString(detail,
+							   "CTEs, subqueries and set-returning functions are not supported by "
+							   "continuous aggregates.");
+		return false;
+	}
+
+	if (query->hasForUpdate || query->hasModifyingCTE)
+	{
+		appendStringInfoString(detail,
+							   "Data modification is not allowed in continuous aggregate view "
+							   "definitions.");
+		return false;
+	}
+
+	if (query->hasRowSecurity)
+	{
+		appendStringInfoString(detail,
+							   "Row level security is not supported by continuous aggregate "
+							   "views.");
+		return false;
+	}
+
+	if (query->groupingSets)
+	{
+		appendStringInfoString(detail,
+							   "GROUP BY GROUPING SETS, ROLLUP and CUBE are not supported by "
+							   "continuous aggregates");
+		appendStringInfoString(hint,
+							   "Define multiple continuous aggregates with different grouping "
+							   "levels.");
+		return false;
+	}
+
+	if (query->setOperations)
+	{
+		appendStringInfoString(detail,
+							   "UNION, EXCEPT & INTERSECT are not supported by continuous "
+							   "aggregates");
+		return false;
+	}
+
+	if (!query->groupClause)
+	{
+		/*
+		 * Query can have aggregate without group by , so look
+		 * for groupClause.
+		 */
+		appendStringInfoString(hint,
+							   "Include at least one aggregate function"
+							   " and a GROUP BY clause with time bucket.");
+		return false;
+	}
+
+	return true; /* Query was OK and is supported. */
+}
+
+static inline Datum
```

Comment:
Nit: compilers usually ignore inline.
```suggestion
static Datum
```